File: go-sdk/kestra_api_client/case_templates_api.go
package kestra_api_client

import "context"

// CaseTemplatesAPI covers /api/v1/{tenant}/case-templates.
type CaseTemplatesAPI struct {
	baseAPI
}

// SearchCaseTemplates returns a page of case templates matching the given filters.
func (a *CaseTemplatesAPI) SearchCaseTemplates(ctx context.Context, tenant string, page, size *int, sort []string, filters []SearchFilter) (*PagedResultsCaseTemplate, error) {
	params := buildQueryParams("page", page, "size", size)
	appendRepeatedParam(params, "sort", sort)
	appendFilterParams(params, filters)
	return doJSON[*PagedResultsCaseTemplate](&a.baseAPI, ctx, "GET", tenantPath(tenant, "case-templates", "search"), nil, params)
}

// CreateCaseTemplate creates a case template.
func (a *CaseTemplatesAPI) CreateCaseTemplate(ctx context.Context, tenant string, request CaseTemplateRequest) (*CaseTemplate, error) {
	return doJSON[*CaseTemplate](&a.baseAPI, ctx, "POST", tenantPath(tenant, "case-templates"), request, nil)
}

// GetCaseTemplate retrieves a case template by id.
func (a *CaseTemplatesAPI) GetCaseTemplate(ctx context.Context, tenant, id string) (*CaseTemplate, error) {
	return doJSON[*CaseTemplate](&a.baseAPI, ctx, "GET", tenantPath(tenant, "case-templates", id), nil, nil)
}

// UpdateCaseTemplate updates a case template.
func (a *CaseTemplatesAPI) UpdateCaseTemplate(ctx context.Context, tenant, id string, request CaseTemplateRequest) (*CaseTemplate, error) {
	return doJSON[*CaseTemplate](&a.baseAPI, ctx, "PUT", tenantPath(tenant, "case-templates", id), request, nil)
}

// DeleteCaseTemplate deletes a case template.
func (a *CaseTemplatesAPI) DeleteCaseTemplate(ctx context.Context, tenant, id string) error {
	return a.doVoidJSON(ctx, "DELETE", tenantPath(tenant, "case-templates", id), nil, nil)
}

File: python/python-sdk/kestrapy/api/case_templates_api.py
from typing import Any, Dict, List, Optional

from kestrapy.base_api import BaseApi
from kestrapy.models.case_templates_controller_api_case_template import (
    CaseTemplatesControllerApiCaseTemplate,
)
from kestrapy.models.paged_results_case_templates_controller_api_case_template import (
    PagedResultsCaseTemplatesControllerApiCaseTemplate,
)
from kestrapy.models.query_filter import QueryFilter


class CaseTemplatesApi(BaseApi):
    """Case templates (`/api/v1/{tenant}/case-templates`).

    A case template pre-fills default severity, assignees, watchers, SLA and
    actions when a case is created from it. Like cases, it is an EE feature gated
    by the ``CASE`` resource, so every route answers 403 when disabled. Request
    bodies are accepted as plain ``Dict[str, Any]``.
    """

    def search_case_templates(
        self,
        tenant: str,
        page: Optional[int] = None,
        size: Optional[int] = None,
        sort: Optional[List[str]] = None,
        filters: Optional[List[QueryFilter]] = None,
    ) -> PagedResultsCaseTemplatesControllerApiCaseTemplate:
        path = self._tenant_path(tenant, "case-templates", "search")
        params = list(self._build_query_params(page=page, size=size).items())
        self._append_repeated_param(params, "sort", sort)
        self._append_filter_params(params, filters)
        return self._json_request(
            "GET", path, PagedResultsCaseTemplatesControllerApiCaseTemplate, params=params
        )

    def create_case_template(
        self, tenant: str, body: Dict[str, Any]
    ) -> CaseTemplatesControllerApiCaseTemplate:
        path = self._tenant_path(tenant, "case-templates")
        return self._json_request("POST", path, CaseTemplatesControllerApiCaseTemplate, body=body)

    def get_case_template(self, tenant: str, id: str) -> CaseTemplatesControllerApiCaseTemplate:
        path = self._tenant_path(tenant, "case-templates", id)
        return self._json_request("GET", path, CaseTemplatesControllerApiCaseTemplate)

    def update_case_template(
        self, tenant: str, id: str, body: Dict[str, Any]
    ) -> CaseTemplatesControllerApiCaseTemplate:
        path = self._tenant_path(tenant, "case-templates", id)
        return self._json_request("PUT", path, CaseTemplatesControllerApiCaseTemplate, body=body)

    def delete_case_template(self, tenant: str, id: str) -> None:
        path = self._tenant_path(tenant, "case-templates", id)
        self._void_request("DELETE", path)
